### PR TITLE
Deprecate DistributedSearchTree

### PR DESCRIPTION
**Pull request by masterleinad**
Addressing 
> Yes, we realized last week that there is inconsistency in the naming. I would argue that `DistributedTree` should be the name we go with, as `DistributedSearchTree` is a tautology (as in, what else you are going to use it for if not search?). I'm not sure about DST abbreviation, as it is not commonly used, unlike BVH.

_Originally posted by @aprokop in https://github.com/arborx/ArborX/pull/394#issuecomment-700057006_

Really just a copy and replace `DistributedSearchTree` -> `DistributedTree` (also for the file names). Additionally, the distributed benchmark's name needed to be slightly changed.
`src/ArborX_DistributedSearchTree.hpp` is still kept for backward-compatibility.

### CODE REVIEW

File: src/ArborX_DistributedTree.hpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  template <typename... Args>
+  void query(Args &&... args) const
+  {
+    DistributedTree<typename DeviceType::memory_space>::query(
+        typename DeviceType::execution_space{}, std::forward<Args>(args)...);
+  }
+};
+
+template <typename MemorySpace, typename Enable = void>
+using DistributedSearchTree[[deprecated]] =
```
Comment (dalg24) [inline src/ArborX_DistributedTree.hpp:210]:
```suggestion
using DistributedSearchTree[[deprecated("Use DistributedTree instead.")]] =
```

**Review by aprokop — COMMENTED** [review]
Needs style check fix, but looks good overall.

**Comment by masterleinad** [comment]
I am still quite sure what the problem with `clang-format` in the `Style` check for this pull request is but I just disabled the problematic part for now.

File: CMakeLists.txt [hunk 2]
```diff
@@ -123,7 +123,6 @@ if(${ARBORX_ENABLE_TESTS})
   endforeach()
   foreach(_file ${ArborX_HEADERS})
     foreach(_macro ${ArborX_MACROS})
-      file(STRINGS src/${_file} _includes_mpi REGEX "mpi.h")
```
Comment (aprokop) [inline CMakeLists.txt:126]: Why did you decide to make this change instead of tracking the place where the issue originated?

Comment (masterleinad) [reply CMakeLists.txt:126]: I don't necessarily see something wrong with including headers that include the distributed implementation even if ArborX has been configured without MPI support.
I would also argue that not all the files that implement the distributed implementation include `mpi.h` directly (which was the reason for the previous commit to fail).

**Comment by dalg24** [comment]
Please elaborate on https://github.com/arborx/ArborX/pull/396/commits/81ad298362e9d0c3ee17f319cda9c54614919fb6

What is it that need fixing?

**Comment by aprokop** [comment]
I failed to reproduce locally the no-MPI build failure from 81ad298.

**Comment by masterleinad** [comment]
@aprokop Here you go.

**Review by aprokop — APPROVED** [review]
Great, thanks!

**Comment by aprokop** [comment]
Merged with standard HIP test failure.

**Comment by aprokop** [comment]
Because HIP failed, I missed the fact that this PR introduced a new warning:
```
This file is deprecated. Use ArborX_DistributedTree.hpp instead!
```
This is coming from the self-contained header test, I believe. What are our options here?

**Comment by masterleinad** [comment]
> This is coming from the self-contained header test, I believe. What are our options here?

I didn't expect the plugin to be more sensitive than `-Werror`... I am not quite sure if the plugin can just be told to ignore these pragma warnings. Other than that, options include guarding the pragma warning by a CMake define (TBB does such things) or explicitly excluding this file from the header check.

**Pull request merged**